source/uwp/Renderer/lib/ActionHelpers.cpp

```diff
@@ -855,6 +907,260 @@ namespace AdaptiveCards::Rendering::Uwp::ActionHelpers
+
+        ComPtr<IUIElement> actionControl;
+        ComPtr<IAdaptiveActionRenderer> renderer;
+        while (!renderer)
+        {
+            HString actionTypeString;
+            RETURN_IF_FAILED(action->get_ActionTypeString(actionTypeString.GetAddressOf()));
+            RETURN_IF_FAILED(actionRegistration->Get(actionTypeString.Get(), &renderer));
+            if (!renderer)
+            {
+                ABI::AdaptiveCards::ObjectModel::Uwp::FallbackType actionFallbackType;
+                RETURN_IF_FAILED(action->get_FallbackType(&actionFallbackType));
+                switch (actionFallbackType)
+                {
+                case ABI::AdaptiveCards::ObjectModel::Uwp::FallbackType::Drop:
+                {
+                    RETURN_IF_FAILED(XamlHelpers::WarnForFallbackDrop(renderContext, actionTypeString.Get()));
+                    return S_OK;
+                }
+
+                case ABI::AdaptiveCards::ObjectModel::Uwp::FallbackType::Content:
+                {
+                    ComPtr<IAdaptiveActionElement> actionFallback;
+                    RETURN_IF_FAILED(action->get_FallbackContent(&actionFallback));
+
+                    HString fallbackTypeString;
+                    RETURN_IF_FAILED(actionFallback->get_ActionTypeString(fallbackTypeString.GetAddressOf()));
+                    RETURN_IF_FAILED(XamlHelpers::WarnForFallbackContentElement(renderContext,
+                                                                                actionTypeString.Get(),
+                                                                                fallbackTypeString.Get()));
+
+                    action = actionFallback;
+                    break;
+                }
+
+                case ABI::AdaptiveCards::ObjectModel::Uwp::FallbackType::None:
+                default:
+                    return E_FAIL;
+                }
+            }
+        }
+
+        RETURN_IF_FAILED(renderer->Render(action.Get(), renderContext, renderArgs, &actionControl));
+
+        if (!buttonVisible)
+        {
+            RETURN_IF_FAILED(actionControl->put_Visibility(Visibility_Collapsed));
+        }
+
+        XamlHelpers::AppendXamlElementToPanel(actionControl.Get(), actionsPanel);
+
+        if (columnDefinitions != nullptr)
+        {
+            // If using the equal width columns, we'll add a column and assign the column
+            ComPtr<IColumnDefinition> columnDefinition = XamlHelpers::CreateABIClass<IColumnDefinition>(
+                HStringReference(RuntimeClass_Windows_UI_Xaml_Controls_ColumnDefinition));
+            RETURN_IF_FAILED(columnDefinitions->Append(columnDefinition.Get()));
+
+            if (buttonVisible)
+            {
+                RETURN_IF_FAILED(columnDefinition->put_Width({1.0, GridUnitType::GridUnitType_Star}));
+            }
+            else
+            {
+                RETURN_IF_FAILED(columnDefinition->put_Width({0, GridUnitType::GridUnitType_Auto}));
```

Comment:
>GridUnitType_Auto [](http://example.com/codeflow?start=79&length=17)

is Auto the same as Pixel?

Reply:
Auto uses the size of the content to size the column. So, in this case, because the only content is a collapsed button, the column size also collapses and ends up with zero width.